state/model.go

```diff
@@ -988,7 +997,16 @@ func (m *Model) destroyOps(ensureNoHostedModels, ensureEmpty bool) ([]txn.Op, er
 			cleanupApplicationsOp,
 			cleanupStorageOp,
 		)
+	} else {
+		if cloudCredential, ok := m.CloudCredential(); ok {
```

Comment:
We typically drop the refcount when *removing* the entity. I think we should do this in State.removeAllModelDocs. That'll also cater for dropping the refcount when exporting a model, or removing a failed import.